tests/smoke/alp_zgetrf.cpp

```diff
@@ -0,0 +1,286 @@
+#endif
+
+#include <alp.hpp>
+#include <alp/algorithms/householder_lu.hpp>
+#include <graphblas/utils/iscomplex.hpp> // use from grb
+#ifdef DEBUG
+#include "../utils/print_alp_containers.hpp"
+#endif
+
+using namespace alp;
+
+using BaseScalarType = double;
+using Orthogonal = structures::Orthogonal;
+using General = structures::General;
+
+#ifdef _COMPLEX
+using ScalarType = std::complex< BaseScalarType >;
+#else
+using ScalarType = BaseScalarType;
+#endif
+
+constexpr BaseScalarType tol = 1.e-10;
+constexpr size_t RNDSEED = 1;
+
+//** generate random rectangular matrix data: complex version */
+template< typename T >
+std::vector< T > generate_rectangular_matrix_data(
+	size_t N,
+	size_t M,
+	const typename std::enable_if<
+		grb::utils::is_complex< T >::value,
+		void
+	>::type * const = nullptr
+) {
+	std::vector< T > data( N * M );
+	for( size_t i = 0; i < N; ++i ) {
+		for( size_t j = 0; j < M; ++j ) {
+			T val( std::rand(), std::rand() );
+			data[ i * M + j ] = val / std::abs( val );
+		}
+	}
+	return data;
+}
+
+//** generate random rectangular matrix data: real version */
+template<
+	typename T
+>
+std::vector< T >  generate_rectangular_matrix_data(
+	size_t N,
+	size_t M,
+	const typename std::enable_if<
+		!grb::utils::is_complex< T >::value,
+		void
+	>::type * const = nullptr
+) {
+	std::vector< T > data( N * M );
+	for( size_t i = 0; i < N; ++i ) {
+		for( size_t j = 0; j < M; ++j ) {
+			data[ i * M + j ] = static_cast< T >( std::rand() ) / RAND_MAX;
+		}
+	}
+	return data;
+}
+
+template<
+	typename D,
+	typename GeneralType,
+	typename GenView,
+	typename GenImfR,
+	typename GenImfC,
+	typename UType,
+	typename UView,
+	typename UImfR,
+	typename UImfC,
+	typename LType,
+	typename LView,
+	typename LImfR,
+	typename LImfC,
+	class Ring = Semiring< operators::add< D >, operators::mul< D >, identities::zero, identities::one >,
+	class Minus = operators::subtract< D >
+	>
+RC check_lu_solution(
+	Matrix< D, GeneralType, alp::Dense, GenView, GenImfR, GenImfC > &H,
+	Matrix< D, LType, alp::Dense, LView, LImfR, LImfC > &L,
+	Matrix< D, UType, alp::Dense, UView, UImfR, UImfC > &U,
+	Vector< size_t > &p,
+	const Ring &ring = Ring(),
+	const Minus &minus = Minus()
+) {
+	RC rc = SUCCESS;
+	const Scalar< D > zero( ring.template getZero< D >() );
+	const Scalar< D > one( ring.template getOne< D >() );
+
+	const size_t m = nrows( H );
+	const size_t n = ncols( H );
+
+#ifdef DEBUG
+	std::cout << " ********************\n";
+	std::cout << " ** check_solution **\n";
+	std::cout << " input:\n";
+	print_matrix( "  H  ", H );
+	print_matrix( "  L  ", L );
+	print_matrix( "  U  ", U );
+	print_vector( "  p  ", p );
+	std::cout << " ********************\n";
+#endif
+
+ 	alp::Matrix< D, GeneralType, alp::Density::Dense > LU( m, n );
+	// LU = L * U
+	rc = rc ? rc : set( LU, zero );
+	rc = rc ? rc : mxm( LU, L, U, ring );
+
+	// until #591 is implemented we use no_permutation_vec explicitly
+	alp::Vector< size_t > no_permutation_vec( n );
```

Comment:
to initialize to id permutation we should use 
`alp::set< alp::descriptors::use_index >( no_permutation_vec, alp::Scalar< size_t >( 0 ) );`
(other use example in `alp::sort` in alp/reference/blas1.hpp)